examples/kubernetes-guestbook/components/Main.scala

```diff
@@ -0,0 +1,131 @@
+import besom.*
+import besom.api.kubernetes
+import besom.api.kubernetes.core.v1.{Service, ServiceArgs}
+import besom.api.kubernetes.core.v1.inputs.*
+import besom.api.kubernetes.apps.v1.{Deployment, DeploymentArgs}
+import besom.api.kubernetes.apps.v1.inputs.*
+import besom.api.kubernetes.meta.v1.*
+import besom.api.kubernetes.meta.v1.inputs.*
+import besom.api.kubernetes.core.v1.enums.ServiceSpecType
+
+case class ServiceDeploymentArgs(
+  image: String,
+  replicas: Int = 1,
+  resources: Option[ResourceRequirementsArgs] = None,
+  ports: List[Int] = List.empty,
+  allocateIPAddress: Option[Boolean] = None,
+  serviceType: Option[ServiceSpecType] = None,
+  env: List[EnvVarArgs] = List.empty
+)
+
+case class ServiceDeployment(ipAddress: Output[Option[String]])(using ComponentBase) extends ComponentResource
```

Comment:
could we check if changing `Output[Option[String]]` to `Input.Optional[String]` leads to better DX? We shouldn't need to wrap with Some at call site then